src/Compilers/CSharp/Portable/Symbols/Source/ThisParameterSymbol.cs

```diff
@@ -40,10 +40,22 @@ public override RefKind RefKind
         {
             get
             {
-                return
-                    ((object)_containingType == null || _containingType.TypeKind != TypeKind.Struct) ? RefKind.None :
-                    ((object)_containingMethod != null && _containingMethod.MethodKind == MethodKind.Constructor) ? RefKind.Out :
-                    RefKind.Ref;
+                if (ContainingType?.TypeKind != TypeKind.Struct)
+                {
+                    return RefKind.None;
+                }
+
+                if (_containingMethod?.MethodKind == MethodKind.Constructor)
```

Comment:
To answer other questions - The `_containingType/_containingMethod` parts did not change. I just refactored the original code. 
Lambdas/LocalFunctions do not have ThisParameter, intentionally. Noone is supposed to get ThisParameter from a lambda symbol.

---
In reply to: [118576819](https://github.com/dotnet/roslyn/pull/19713#discussion_r118576819) [](ancestors = 118576819,118565720,118552148)